setup/gcp/installer/forseti_installer.py

```diff
@@ -0,0 +1,291 @@
+                                                          'reader',
+                                                          self.config.timestamp,
+                                                          self.project_id)
+
+    def generate_bucket_name(self):
+        """Generate GCS bucket name.
+
+        Returns:
+            str: Name of the GCS bucket
+        """
+        return DEFAULT_BUCKET_FMT_V2.format(self.project_id,
+                                            self.config.template_type,
+                                            self.config.timestamp)
+
+    @abstractmethod
+    def get_deployment_values(self):
+        """Get deployment values
+
+        Returns:
+            dict: A dictionary of values needed to generate
+                the forseti deployment template
+        """
+        return {}
+
+    @abstractmethod
+    def get_configuration_values(self):
+        """Get configuration values
+
+        Returns:
+            dict: A dictionary of values needed to generate
+                the forseti configuration file
+        """
+        return {}
+
+    def generate_deployment_templates(self):
+        """Generate deployment templates.
+
+        Returns:
+            str: Deployment template path
+        """
+        print('Generate Deployment Manager templates...')
+
+        deploy_values = self.get_deployment_values()
+
+        deploy_tpl_path = generate_deployment_templates(
+            self.config.template_type,
+            deploy_values,
+            self.config.datetimestamp)
+
+        print('\nCreated a deployment template:\n    %s\n' %
+              deploy_tpl_path)
+        return deploy_tpl_path
+
+    def generate_forseti_conf(self):
+        """Generate Forseti conf file.
+
+        Returns:
+            str: Forseti configuration file path
+        """
+        # Create a forseti_conf_$TIMESTAMP.yaml config file with
+        # values filled in.
+        # forseti_conf_server.yaml in file
+        print('\nGenerate forseti_conf_{}_{}.yaml...'
+              .format(self.config.template_type, self.config.datetimestamp))
+
+        conf_values = self.get_configuration_values()
+
+        forseti_conf_path = generate_forseti_conf(self.config.template_type,
+                                                  conf_values,
+                                                  self.config.datetimestamp)
+
+        print('\nCreated forseti_conf_{}_{}.yaml config file:\n    {}\n'.
+              format(self.config.template_type,
+                     self.config.datetimestamp,
+                     forseti_conf_path))
+        return forseti_conf_path
+
+    def post_install_instructions(self, deploy_success, deployment_name,
+                                  deploy_tpl_path, forseti_conf_path,
+                                  bucket_name):
+        """Show post-install instructions
+
+        Print link for deployment manager dashboard
```

Comment:
nit: add period at the end of the sentence

Reply:
Updated.